plugins/modules/proxmox_backup_info.py

```diff
@@ -0,0 +1,234 @@
+      returned: on success
+      type: str
+    next-run:
+      description: The next backup time.
+      returned: on success
+      type: str
+    schedule:
+      description: The backup job schedule.
+      returned: on success
+      type: str
+    storage:
+      description: The backup storage location.
+      returned: on success
+      type: str
+    vm_name:
+      description: The VM name.
+      returned: on success
+      type: str
+    vmid:
+      description: The VM vmid.
+      returned: on success
+      type: int
+"""
+
+from datetime import datetime
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible_collections.community.general.plugins.module_utils.proxmox import (
+    proxmox_auth_argument_spec, ProxmoxAnsible, HAS_PROXMOXER, PROXMOXER_IMP_ERR)
+
+
+class ProxmoxBackupInfoAnsible(ProxmoxAnsible):
+
+    # Get all backup information
+    def getJobsList(self):
+        try:
+            backupJobs = self.proxmox_api.cluster.backup.get()
+        except Exception as e:
+            self.module.fail_json(msg="Getting backup jobs failed: %s" % e)
+        return backupJobs
+
+    # Get VM information
+    def getVmsList(self):
+        try:
+            vms = self.proxmox_api.cluster.resources.get(type='vm')
+        except Exception as e:
+            self.module.fail_json(msg="Getting VMs info from cluster failed: %s" % e)
+        return vms
+
+    # Get all backup information by VM id and VM name
+    def vmsBackupInfo(self):
+        backupList = self.getJobsList()
+        vmInfo = self.getVmsList()
+        bkInfo = []
+        for backupItem in backupList:
+            nextrun = datetime.fromtimestamp(backupItem['next-run'])
+            vmids = backupItem['vmid'].split(',')
+            for vmid in vmids:
+                for vm in vmInfo:
+                    if vm['vmid'] == int(vmid):
+                        vmName = vm['name']
+                        break
+                bkInfoData = {'id': backupItem['id'],
+                              'schedule': backupItem['schedule'],
+                              'storage': backupItem['storage'],
+                              'mode': backupItem['mode'],
+                              'next-run': nextrun.strftime("%Y-%m-%d %H:%M:%S"),
+                              'enabled': backupItem['enabled'],
+                              'bktype': backupItem['type'],
+                              'vmid': vmid,
+                              'vm_name': vmName}
+                bkInfo.append(bkInfoData)
+        return bkInfo
+
+    # Get proxmox backup information for a specific VM based on its VM id or VM name
+    def specificVmBackupInfo(self, vm_name_id):
+        fullBackupInfo = self.vmsBackupInfo()
+        vmBackupJobs = []
+        for vm in fullBackupInfo:
+            if (vm["vm_name"] == vm_name_id or int(vm["vmid"]) == vm_name_id):
+                vmBackupJobs.append(vm)
+        return vmBackupJobs
```

Comment:
I don't think this is right. If `vm_name_id="100"` (str), then `int(vm["vmid"]) == vm_named_id` is always going to be `False`, even then `int(vm["vmid"]) == 100`.
```python
>>> 3 == "3"
False
```

Additionally, there is the naming convention (camelCase vs. snake_case), and the more pythonic way of doing this would be:
```suggestion
        return [vm for vm in full_backup_info if vm["vm_name"] == vm_name_id or int(vm["vmid"]) == vm_name_id]
```
(it won't work like that because of the type issue mentioned above - needs adjustments)

Not goin to be a huge problem here, but from a performance point of view it is much better to do the test outside the loop than inside it.
```
if is int
  return [... if vm[vmid] == param]
if is str
  return [... if vm[name] == param]
```
Or, make two separate functions, one for the ID another for the name, which is cleaner to read, IMHO.